### PR TITLE
Fix id type issue for update

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @ritch  or @bajtos 

The PR fixes https://github.com/strongloop/loopback/issues/1058

### CODE REVIEW

File: test/memory.test.js [hunk 1]
```diff
@@ -7,11 +7,11 @@ var async = require('async');
 var should = require('./init.js');
 var Memory = require('../lib/connectors/memory').Memory;
 
-describe('Memory connector', function () {
+describe('Memory connector1', function() {
```
Comment (bajtos) [inline test/memory.test.js]: why `connector`?

File: test/memory.test.js [hunk 2]
```diff
@@ -19,82 +19,113 @@ describe('Memory connector', function () {
...
+              }
+              readModels(function(err, json) {
+                if (err) {
+                  return done(err);
+                }
+                assert.equal(Object.keys(json.models.User).length, 2);
+                var user = JSON.parse(json.models.User[ids[1]]);
+                assert.equal(user.name, 'John');
+                assert(user.id === ids[1]);
+                User.update({id: user.id}, {name: 'John1'},
```
Comment (bajtos) [inline test/memory.test.js]: This test is already too long and testing way too many things.

Don't add your new test code into this existing test as it makes things even worse. Please create a new test focused only on the use case you are fixing.

**Comment by raymondfeng** [comment]
@bajtos Tests are refactored.

**Comment by bajtos** [comment]
Thanks. It would be even better if the tests don't depend on each other, but we can improve that later.

LGTM.

**Comment by pulkitsinghal** [comment]
used the master v2.15.0 and it works

**Pull request merged**